### ISSUE TITLE
[Linux] In which container/context does the Deployment Script run?

### ISSUE THREAD

**Issue by blueelvis** [projectkudu/kudu#2677]
### Repro steps.
> Create a web app for container
> use the default deployment scripts generated by kuduscript
> modify the kuduscript to use a program installed in the container 

### The log/error given by the failure.
> Unable to find such command

I tried running a program as part of the deployment script (drush) but the deployment log said that it couldn't find the program. Could you please describe how and where the deployment script is executed?

**Comment by nickwalkmsft** [comment]
Deployment scripts run in the Kudu container; this is by design.

We don't really have a documented workflow for it, but one thing you could try is using ssh to connect from the Kudu container to your app container to run a command during your script. Your app container needs to be running ssh server; you can find the ip address to connect to in the id_addr file in /home/site.

**Comment by blueelvis** [comment]
@nickwalkmsft - that sounds an interesting workflow. is it possible we could probably have a small chat about this?

**Comment by blueelvis** [comment]
So I just checked out the `ipaddr_0` file and it gave me - `172.19.0.3` . A few questions surrounding this -

1. Is this IP static?
2. If not, when does it change? Reboot of the app container? Scale up/down of App Service Plan?
3. How about scaling out the app service plan?


Sorry to sound very excited but I have been working with the web app for containers and really excited about what could be achieved! Thanks!

**Comment by blueelvis** [comment]
I just verified that the IP is not static. Assuming that the Kudu container has access to the `/home` folder and `WEBSITES_ENABLE_APP_SERVICE_STORAGE` is set to `True`, I can do something as below right?

1. Read the ipaddr_0 file
2. Use the results from above to establish SSH connection (Password is always `Docker!` and username `root` as per Azure Documentation for Web App for containers)
3. Run commands via SSH and profit?

I am still not sure about how scaling out will work out. I have scaled out my App Service Plan but now I am unable to SSH into the website at all and it has been almost 30 minutes since the scale out operation...
